### PR TITLE
fix some memory leaks in the NBD server and enable asan for the tests

### PR DESCRIPTION
**Pull request by jluebbe**
Most of the nbd asan tests are not running in the CI, as we're not using qemu there, yet.

### CODE REVIEW

File: src/nbd.c [hunk 1]
```diff
@@ -556,12 +556,12 @@ static void start_configure(struct RaucNBDContext *ctx, struct RaucNBDTransfer *
...
 	CURLMcode mcode = 0;
-	g_autofree guint8 *data = g_malloc(xfer->request.len);
-	g_autoptr(GVariant) v = NULL;
-	GVariantDict dict;
 
 	/* only read from the client on the first try */
 	if (!ctx->url) {
+		g_autofree guint8 *data = g_malloc(xfer->request.len);
+		g_autoptr(GVariant) v = NULL;
+		g_auto(GVariantDict) dict = G_VARIANT_DICT_INIT(NULL);
```
Comment (ejoerns) [inline src/nbd.c:564]: Maybe note that this changes from stack-allocated to heap-allocated and that the former variant lacked the `g_variant_dict_clear()` call?

Also, the commit contains an unrelated newline change and unrelated code move. But since the result is still clear enough to read I'd be fine with that.

File: src/nbd.c [hunk 2]
```diff
@@ -951,6 +951,10 @@ gboolean r_nbd_run_server(gint sock, GError **error)
 
 	res = TRUE;
 out:
+	g_clear_pointer(&ctx.url, g_free);
```
Comment (ejoerns) [inline src/nbd.c:957]: That's quite close to justifying a cleanup method for `RaucNBDContext` :wink: 
But fine first of all.

**Review by ejoerns — COMMENTED** [review]
Great to have the nbd c module asan-tested \o/

File: src/nbd.c (hunk 1 above)
Comment (jluebbe) [reply src/nbd.c:564]: This is still stack-allocated, which is what `g_auto` is used for.

Comment (ejoerns) [reply src/nbd.c:564]: Ouh, right. My fault.

**Review by ejoerns — APPROVED** [review]
